Current Discount Implementation

Dennybah opened the pull request:
Current Discount Implemented, should not interfere with any previous code

Dennybah committed:
Current Discount Implementation

BryanCarty commented on Book-eCommerce-Store/Services/OrdersService/OrdersService.cs:
```diff
@@ -69,13 +74,55 @@ public async Task<Response> CreateOrder(CreateOrderDTO newOrder)
+                    DiscountName = mappedToPurchase.Name;
+                    DiscountQuantityCount = mappedToPurchase.Quantity;
+                    DiscountCategory = (int)mappedToPurchase.ProductCategory;
                 }
```
Think you accidently placed the code to calculate the discount outside the foreach loop in CreateOrder. At the moment it will only apply a discount to the last item in the list if applicable because DiscountName, DiscountQuantity and DiscountCategory will keep getting overwritten.

Dennybah replied:
I should have it sorted now, if something else needs to be changed lemme know

Dennybah committed:
Fixed create order, discount no longer applies to last item

Added additional checks eg needing to have a quantity of 1 of benefit from winter discount (having 1 of any book in your order)

Dennybah committed:
Revert "Fixed create order, discount no longer applies to last item"

This reverts commit f212c2bf0fc510012210ca92d21373bb6141721a.

Dennybah committed:
Fixed create order, discount no longer applies to last item

Uploaded to the correct location this time
